eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php

```diff
@@ -38,6 +51,79 @@ public function setOutput($output)
         $this->output = $output;
     }
 
+    /**
+     * Import Schema using event-driven Schema Builder API from eZ Systems DoctrineSchema Bundle.
+     *
+     * If you wish to extend schema, implement your own EventSubscriber
+     *
+     * @see \EzSystems\DoctrineSchema\API\Builder\SchemaBuilderEvent
+     * @see \EzSystems\PlatformInstallerBundle\Event\Subscriber\BuildSchemaSubscriber
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function importSchema()
+    {
+        // note: schema is built using Schema Builder event-driven API
+        $schema = $this->schemaBuilder->buildSchema();
+        $databasePlatform = $this->db->getDatabasePlatform();
+        $queries = array_merge(
+            $this->getDropSqlStatementsForExistingSchema($schema, $databasePlatform),
+            // generate schema DDL queries
+            $schema->toSql($databasePlatform)
+        );
+
+        $queriesCount = count($queries);
+        $this->output->writeln(
+            sprintf(
+                '<info>Executing %d queries on database <comment>%s</comment> (<comment>%s</comment>)</info>',
+                $queriesCount,
+                $this->db->getDatabase(),
+                $this->db->getDatabasePlatform()->getName()
+            )
+        );
+        $progressBar = new ProgressBar($this->output);
+        $progressBar->start($queriesCount);
+
+        try {
+            $this->db->beginTransaction();
+            foreach ($queries as $query) {
+                $this->db->exec($query);
+                $progressBar->advance(1);
+            }
+            $this->db->commit();
+        } catch (DBALException $e) {
+            $this->db->rollBack();
+            throw $e;
+        }
+
+        $progressBar->finish();
+        $this->output->writeln('');
```

Comment:
Maybe instead of displaying nothing, we can inform a user that everything ended successfully

Reply:
> Maybe instead of displaying nothing, we can inform a user that everything ended successfully

@mikadamczyk we could, but this is not about that. ProgressBar apparently doesn't do CRLF on `finish()` so every command for not misaligned output needs to be finished with this. I'll add inline doc.

Reply:
@mikadamczyk see the PR description - I've attached screenshot of the installer operation. Extra info after the first ProgressBar doesn't make sense here, so I just kept inline comment.

Reply:
💯